### ISSUE TITLE
no-else-return can't catch nested issues

### ISSUE THREAD

**Issue by elwayman02** [eslint/eslint#1369]
The following code snippet will throw one error from no-else-return, when it should throw two:

```
if (val) {
    // do stuff
    if (val2) {
        return stuff;
    } else {
        return otherStuff;
    }
} else {
    return thirdStuff;
}
```

Technically, I think fixing the first error will expose the second error when the rule is next run, but now you're requiring multiple runs of eslint to catch a problem that should be detected from the beginning.  Is there a way to detected nested violations of this error without significantly impacting the performance?
